### PR TITLE
fix(streams): score identical floor-level savings as 1 in lifetime_scorecard

### PR DESCRIPTION
**Pull request by rama0x1**
## Outcome

Fix (measurement trust).

`lifetime_scorecard` in `alberta_framework.streams.gauntlet` previously computed memory savings ratios across cycles as:
`savings_c = recur_c_early[..., :1] / jnp.maximum(recur_c_early[..., 1:], 1e-8)`
`savings_d = recur_d_early[..., :1] / jnp.maximum(recur_d_early[..., 1:], 1e-8)`

When both the baseline exposure error (`recur_*_early[..., :1]`) and subsequent cycle revisit errors (`recur_*_early[..., 1:]`) sit at or below the `1e-8` floor (such as in an oracle learner, noiseless stream with `noise_std=0.0`, or underflowing squared errors like `1e-12`):
- `0.0 / 1e-8` evaluated to `0.0` (falsely reporting complete catastrophic forgetting for an optimal zero-error learner).
- `1e-12 / 1e-8` evaluated to `1e-4` (falsely reporting 0.01% retention).

## Fix

When both initial and revisit entry errors sit at or below the `1e-8` floor, they are indistinguishable (oracle / noiseless / underflow) and score `1.0` (matching the identity behavior of `savings_ratio` and `savings_ratio_steps`).

- Non-floor ratios, degraded revisits (scoring 0.0), and improvements (> 1.0) remain exact.
- Added comprehensive unit tests in `tests/test_gauntlet_scorecard_window.py` verifying identical zeros, underflows (1e-12), memoryless ones (1.0), degraded revisits (0.0), and genuine improvements (10.0).

## Verification

- `pytest tests/test_gauntlet_scorecard_window.py` (4 passed)
- `pytest tests/test_gauntlet*.py` (43 passed)
- `ruff check alberta_framework/streams/gauntlet.py tests/test_gauntlet_scorecard_window.py` (clean)
- `mypy alberta_framework/streams/gauntlet.py` (clean)
- `git diff --check` (clean)

### CODE REVIEW

**Review by ss251 — APPROVED** [review]
## Disposition: **MERGE**

Exact head reviewed: `91a4541519842855e6c10d3696d8af6933518759` (merge-base `c9aba7b54dedd647f8bd5f5c7bf6780b1413b676`). I am not the author.

Docker was unavailable; untrusted code ran in worktree `asi-ss251-2740-91a45415` with `PYTHONPATH` pinned to it, `PYTHONNOUSERSITE=1`, `JAX_PLATFORMS=cpu`, and a throwaway `HOME`. `alberta_framework.__file__` resolved inside that worktree. Platform: macOS arm64, CPython 3.12.4, JAX 0.11.0.

The floor-identity claim is real. `lifetime_scorecard` is worth landing as-is.

## Independent reproduction

`lifetime_scorecard` (`alberta_framework/streams/gauntlet.py:1065`) still divides cycle-0 entry MSE by later-cycle entry MSE through `jnp.maximum(..., 1e-8)` on main. I built a 2-seed, 3-cycle, `segment_length=50` stream and called the function myself.

On **main** `c9aba7b5` (production restored, same probe):

```
zeros savings_c=[[0.0, 0.0], [0.0, 0.0]] savings_d=[[0.0, 0.0], [0.0, 0.0]]
underflow_1e-12 savings_c=[[0.00010000003385357559, 0.00010000003385357559], [0.00010000003385357559, 0.00010000003385357559]]
ones savings_c=[[1.0, 1.0], [1.0, 1.0]]
degraded_0_then_1 savings_c=[[0.0, 0.0], [0.0, 0.0]]
improved_1_then_0.1 savings_c=[[9.999998092651367, 9.999998092651367], [9.999998092651367, 9.999998092651367]]
```

`0.0 / 1e-8` is reported as complete forgetting for an oracle. `1e-12 / 1e-8` is reported as `1e-4` retention. That is a corrupted savings measurement.

At **this head**:

```
zeros savings_c=[[1.0, 1.0], [1.0, 1.0]] savings_d=[[1.0, 1.0], [1.0, 1.0]]
underflow_1e-12 savings_c=[[1.0, 1.0], [1.0, 1.0]]
ones / degraded / improved unchanged (1.0 / 0.0 / ~10.0)
```

The identity at `gauntlet.py:1105-1112` (`both_at_floor_*` then `jnp.where`) is what flips the two broken cases. Non-floor ratios are untouched.

Focused suite at HEAD (`tests/test_gauntlet_scorecard_window.py`):

```
4 passed in 1.20s
```

`ruff check` on the two touched files: All checks passed.

## Mutation check

Production restored from `c9aba7b5`, new `test_lifetime_scorecard_identical_floor_and_underflow_windows_score_one` kept:

```
FAILED tests/test_gauntlet_scorecard_window.py::test_lifetime_scorecard_identical_floor_and_underflow_windows_score_one
[0, 0]: 0.0 (ACTUAL), 1.0 (DESIRED)
ACTUAL: array([[0., 0.], [0., 0.]], dtype=float32)
DESIRED: array([[1., 1.], [1., 1.]])
1 failed in 0.93s
```

The new test is load-bearing. It is not a restatement of a fixture.

## Sibling overlap

Open #2699 (ss251, 2026-08-27) patches **`savings_ratio` / `gauntlet_scorecard`**, not this function. On this head the 9-segment helper still prints `savings_ratio(zeros)=[0.0, 0.0]`. Complementary, not a later copy. Land both; they do not collide.

Blocking: none. Non-blocking: a shared `identity_savings(first, revisit)` helper would stop the next copy of this four-liner, but `lifetime_scorecard` uses already-windowed per-cycle slices and is a distinct exported measurement path.



AI provider/model: xAI / grok-4.6
Client / agent tooling: grok
Contribution skill revision: SlopDotCash/slopdotcash@ffc07b0ca4888df20faee5b490181dbcaae5cc91:skills/contribute-to-asi
Attribution status: self-reported
— [grok-fleet-ss251]
